RFS/src/test/java/com/rfs/integration/WorkerCompleteAfterErrorTest.java

```diff
@@ -0,0 +1,118 @@
+package com.rfs.integration;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Test case to verify worker will not deadlock and complete all work given failures
+ * during critical migration workflows.
+ */
+@Slf4j
+public class WorkerCompleteAfterErrorTest {
```

Comment:
Also - in order for these tests to actually be interesting, we'll want the nodes to be doing the *actual work* of the phase they're slated for execution in to test the ability of other nodes to pick up where the one before them left off.

Reply:
> the ConcurrentAllWorkerFailure test being triggered when every node hits the "index setup" phase is a bit nonsensical because only one node will get that work

I'm not sure where nonsensical comes from, the workers don't actually know how to coordinate work, they just do the work that looks available to them.  It could be possible that the target cluster was in a base state and RFS workers keep starting and failing, until it recovered.  So long as the expense of writing and running that test isn't large I'd rather be more thorough - however if either of these cases are no longer true then its a good candidate to remove a failure condition.

Reply:
After we've implemented a first two core scenario, it will help us understand if its more trouble than its worth to get this kind of validation.  What do you think?

Reply:
As I understand the current paradigm, our parameter matrix for these tests has a fair number of redundant, uninteresting or unnecessarily time consuming test paths - like I pointed out with the `ConcurrentAllWorkerFailure` situation when doing the `INDEX_SETUP` injected failure.  To play out that example, the overwhelming majority of the value is in ensuring that another node picks up the work after the first node's failure.  Ensuring that the N+2, N+3, N+X scenarios behave the same way just makes our tests run longer for little additional value.